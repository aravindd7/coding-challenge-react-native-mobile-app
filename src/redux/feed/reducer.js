import {
  FEED_UPDATE__REQUEST,
  FEED_UPDATE__FAIL,
  FEED_UPDATE__SUCCESS,
  FEED_SECTION_CHANGE
} from "./actions";

const initialState = {
  section: "new", // new | top | hot | controversial
  state: "request", // request | fail | success
  error: undefined // Error message
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case FEED_UPDATE__REQUEST:
      /* Reset state before fetch request */
      return { ...state, state: "request" };

    case FEED_UPDATE__FAIL:
      /* Update Fail fields */
      if (action.payload.error === undefined) return state;
      return { ...state, state: "fail", error: action.payload.error };

    case FEED_UPDATE__SUCCESS:
      /* Update Success fields */
      return { ...state, state: "success" };

    case FEED_SECTION_CHANGE:
      /* Chage Section */
      return { ...state, section: action.payload.section };

    default:
      return state;
  }
};

export default reducer;
