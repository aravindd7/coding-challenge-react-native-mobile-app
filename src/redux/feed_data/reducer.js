import { FEED_DATA__UPDATE } from "./actions";

const initialState = []; // Array of news

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case FEED_DATA__UPDATE:
      if (action.payload.data === undefined) return state;
      return action.payload.data;

    default:
      return state;
  }
};

export default reducer;
