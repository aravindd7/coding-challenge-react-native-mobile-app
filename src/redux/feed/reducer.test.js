import {
  FEED_UPDATE__REQUEST,
  FEED_UPDATE__FAIL,
  FEED_UPDATE__SUCCESS,
  FEED_SECTION_CHANGE
} from "./actions";

import reducer from "./reducer";

it("Reducer other action", async () => {
  const action = {
    type: "NO-ACTION"
  };
  const state = reducer(undefined, action);
  expect(state).toMatchSnapshot();
});

it("Reducer action: FEED_UPDATE__REQUEST", async () => {
  const action = {
    type: FEED_UPDATE__REQUEST,
    payload: {}
  };
  const state = reducer(undefined, action);
  expect(state).toMatchSnapshot();
});

it("Reducer action: FEED_UPDATE__FAIL", async () => {
  const action = {
    type: FEED_UPDATE__FAIL,
    payload: {}
  };
  const state = reducer(undefined, action);
  expect(state).toMatchSnapshot();
});

it("Reducer action: FEED_UPDATE__FAIL with error", async () => {
  const action = {
    type: FEED_UPDATE__FAIL,
    payload: { error: "error message" }
  };
  const state = reducer(undefined, action);
  expect(state).toMatchSnapshot();
});

it("Reducer action: FEED_UPDATE__SUCCESS", async () => {
  const action = {
    type: FEED_UPDATE__SUCCESS,
    payload: {}
  };
  const state = reducer(undefined, action);
  expect(state).toMatchSnapshot();
});

it("Reducer action: FEED_SECTION_CHANGE", async () => {
  const action = {
    type: FEED_SECTION_CHANGE,
    payload: { section: "top" }
  };
  const state = reducer(undefined, action);
  expect(state).toMatchSnapshot();
});
