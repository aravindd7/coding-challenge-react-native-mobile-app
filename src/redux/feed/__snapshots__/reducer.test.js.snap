// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer action: FEED_SECTION_CHANGE 1`] = `
Object {
  "error": undefined,
  "section": "top",
  "state": "request",
}
`;

exports[`Reducer action: FEED_UPDATE__FAIL 1`] = `
Object {
  "error": undefined,
  "section": "new",
  "state": "request",
}
`;

exports[`Reducer action: FEED_UPDATE__FAIL with error 1`] = `
Object {
  "error": "error message",
  "section": "new",
  "state": "fail",
}
`;

exports[`Reducer action: FEED_UPDATE__REQUEST 1`] = `
Object {
  "error": undefined,
  "section": "new",
  "state": "request",
}
`;

exports[`Reducer action: FEED_UPDATE__SUCCESS 1`] = `
Object {
  "error": undefined,
  "section": "new",
  "state": "success",
}
`;

exports[`Reducer other action 1`] = `
Object {
  "error": undefined,
  "section": "new",
  "state": "request",
}
`;
